app/models/image.rb

```diff
@@ -29,10 +50,132 @@ def self.thumbnail
 
   # for backwards compatibility, null statuses are considered final
   def fixerable_final?
-    status.nil? || status == COMPLETE
+    status != NOTFOUND && status != UPLOADED
```

Comment:
Yes, though I have added a few more to the list since this ([FAILED, NOTFOUND, UPLOADED, INVALID]) - it's not clear to me why you would want FAILED and INVALID to not be treated as fixerable_final, since all that gets used for is to determine whether to send people to the old URL or not, but I have updated because the tests demanded it.

Reply:
Yeah, I'm fine with either way.  Probably just an oversight - FAILED and INVALID being in the "final location" makes sense.